### ISSUE TITLE
Can't insert nbsp character

### ISSUE THREAD

**Issue by erfannoury** [Microsoft/vscode#6077]
- VSCode Version: 

```
Version 1.0.0
Commit fa6d0f03813dfb9df4589c30121e9fcffa8a8ec8
Date 2016-04-13T14:08:36.599Z
Shell 0.35.6
Renderer 45.0.2454.85
Node 4.1.1
```
- OS Version:

```
Windows Version 10.0.10586
```

Steps to Reproduce:
1. Open an existing text file, or create a new file.
2. Change keyboard language to Persian.
3. Press `Ctrl + Shift + 2` to insert nbsp.

Prior versions of vscode inserted nbsp in the text without any problem, However the latest version doesn't seem to be able to insert nbsp into the text.

**Comment by alexandrudima** [comment]
@erfannoury That is indeed a default shortcut we ship with. Can you please try the following:
* Open `File > Preferences > Keyboard Shortcuts`
* In the `keybindings.json` add the following rules:
```
// Place your key bindings in this file to overwrite the defaults
[
    { "key": "ctrl+shift+1", "when": "editorTextFocus" },
    { "key": "ctrl+shift+2", "when": "editorTextFocus" },
    { "key": "ctrl+shift+3", "when": "editorTextFocus" }
]
```

**Comment by erfannoury** [comment]
@alexandrudima Thank you very much. This solved the problem. 
So according to this solution, Keyboard Shortcuts changed in this version.
I'm closing the issue.

**Comment by erfannoury** [comment]
Sorry for reopening this issue, but it has surfaced again in this version.
```
Version 1.1.1
Commit def9e32467ad6e4f48787d38caf190acbfee5880
Date 2016-05-13T13:29:20.776Z
Shell 0.37.6
Renderer 49.0.2623.75
Node 5.10.0
```
Even though I have applied the changes you proposed before and indeed they worked on the previous version, they are not functional in this version and I can't insert the nbsp character.

**Comment by alexandrudima** [comment]
@erfannoury I don't think we did any changes in this area.

Is it possible that your `keybindings.json` got lost? i.e. when you open `File > Preferences > Keyboard Shortcuts` do you still see these rules:

```
// Place your key bindings in this file to overwrite the defaults
[
    { "key": "ctrl+shift+1", "when": "editorTextFocus" },
    { "key": "ctrl+shift+2", "when": "editorTextFocus" },
    { "key": "ctrl+shift+3", "when": "editorTextFocus" }
]
```

Is it possible that you have a syntax error in `keybindings.json`? That might cause them not to be applied.

**Comment by erfannoury** [comment]
@alexandrudima Yes. I rechecked everything and my `keybinding.json` is there and it contains exactly what you have provided. However, I still have the problem.

**Comment by alexandrudima** [comment]
I believe this is perhaps a change in the keyboard layout shipped by Windows.

Using the Persian keyboard layout, `ctrl+shift+2` does not insert anything for me also when using Notepad, therefore I don't believe this is something specific to VS Code:

![6077](https://cloud.githubusercontent.com/assets/5047891/26282946/11adf034-3e1d-11e7-886c-bfabfd07fece.gif)

**Comment by erfannoury** [comment]
Instead of the `Ctrl + Shift + 2`, I switched to the Standard Persian Layout and it is now working.
For everyone else that wants to know how nbsp is used in Persian, when we want to append a prefix, we use nbsp instead of space in some occasions. With space we have `می توانم`, while it should be written as `می‌توانم`. Without space (and nbsp) it would become `میتوانم` which is not desirable.

**TLDR**: use the Standard Layout for Persian keyboard.
